rmgpy/molecule/group.py

```diff
@@ -277,6 +277,46 @@ def __losePair(self, pair):
         # Set the new atom types, removing any duplicates
         self.atomType = list(set(atomType))
 
+    def __gainCharge(self, charge):
+        """
+        Update the atom charge as a result of applying a GAIN_CHARGE action,
+        where `charge` specifies the number of postive charges to gain.
+        """
+
+        chargeNum = []
+
+        if not self.charge:
+            raise ActionError('Unable to update GroupAtom due to GAIN_CHARGE action: Unknown charge status for GroupAtom "{0}".'.format(self.atomType))
+        else:
+            for x in self.charge:
+                chargeNum.append(x + charge)
+            self.charge = chargeNum
+
+        # Have not update __gainCharge to the molecule.atomtype
```

Comment:
I think we should add the gainCharge and looseCharge actions to atomType.py where appropriate, even if in most (all?) cases the atomType remains the same. We could discuss this in person if you'd like.